builder/build.go

```diff
@@ -11,16 +11,22 @@ import (
 	"path/filepath"
 	"strconv"
 	"strings"
+	"sync"
 
 	"github.com/tinygo-org/tinygo/compileopts"
 	"github.com/tinygo-org/tinygo/compiler"
 	"github.com/tinygo-org/tinygo/goenv"
 	"github.com/tinygo-org/tinygo/interp"
 )
 
+// Due to some problems with embedded Clang and LLD, we cannot run those in
+// parallel, or in parallel with Go compiles. Therefore, we put a lock around
+// builds and run everything else in parallel.
+var buildLock sync.RWMutex
+
 // Build performs a single package to executable Go build. It takes in a package
 // name, an output path, and set of compile options and from that it manages the
-// whole compilation process.
+// whole compilation process. It is safe to be called concurrently.
```

Comment:
We may need to clarify that the caller cannot use LLVM elsewhere while this is running.

Reply:
Fixed.